File: exercise_136.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hashMap = {}
        for number in nums:
            if number not in hashMap:
                hashMap[number] = 1
            else:
                del(hashMap[number])
        return list(hashMap.keys())[0]

File: java/hashsetOrHashmap/exercise_136.java
class Solution {
    public int singleNumber(int[] nums) {
        HashSet<Integer> seen = new HashSet<>();
        for(int i=0;i<nums.length;i++){
            if(seen.contains(nums[i])){
                seen.remove(nums[i]);
            }else{
                seen.add(nums[i]);
            }
        }
        for (Integer number : seen) {
            return number;
        }
        return 0;
    }
}